north/src/runtime/mod.rs

```diff
@@ -121,71 +170,63 @@ pub async fn run(config: &Config) -> Result<(), Error> {
         .collect::<Vec<Name>>();
     for app in &autostart_apps {
         info!("Autostarting {}", app);
-        if let Err(e) = state.start(&app).await {
-            warn!("Failed to start {}: {}", app, e);
-        }
+        state.start(&app).await.ok();
     }
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    enable_signaled_shutdown(event_tx.clone()).await;
-
     // Initialize console
     let console = console::Console::new(&config.console_address, &event_tx);
     // Start to listen for incoming connections
     console.listen().await?;
 
+    // Wait for a external shutdown request
+    let shutdown_tx = event_tx.clone();
+    task::spawn(async move {
+        stop.await.ok();
+        shutdown_tx.send(Event::Shutdown).await.ok();
+    });
+
     // Enter main loop
-    while let Some(event) = event_rx.recv().await {
-        match event {
+    loop {
+        let result = match event_rx.recv().await.unwrap() {
```

Comment:
We should not use unwraps in the code base. If we really want to panic, we should explicitly use `panic!("bla")` or s.th. like that.